### PR TITLE
Implement merging of revisit records through proxy

### PR DESCRIPTION
**Pull request by Avokadoen**
Fixed history of #54

### CODE REVIEW

**Comment by Avokadoen** [comment]
Blocked by [gowarc #56](https://github.com/nlnwa/gowarc/pull/56)

File: cmd/serve/serve.go [hunk 1]
```diff
@@ -128,6 +131,15 @@ func serveCmd(cmd *cobra.Command, args []string) error {
 			excludes = append(excludes, re)
 		}
 	}
+	// parse proxy url
+	var proxyUrl *url.URL = nil
```
Comment (maeb) [inline cmd/serve/serve.go:135]:
```suggestion
	var proxyUrl *url.URL
```
Pointers are nil by default.

File: internal/loader/filestorageloader.go [hunk 2]
```diff
@@ -19,17 +19,19 @@ package loader
...
+	"github.com/nlnwa/gowarc"
+	"github.com/rs/zerolog/log"
 )
 
 type FileStorageLoader struct {
 	FilePathResolver func(fileName string) (filePath string, err error)
 }
 
 func (f *FileStorageLoader) Load(ctx context.Context, storageRef string) (record gowarc.WarcRecord, err error) {
+	log.Debug().Msg("file storage load")
```
Comment (maeb) [inline internal/loader/filestorageloader.go:34]:
```suggestion
```
Stray debug log message..

File: internal/loader/loader.go [hunk 3]
```diff
@@ -53,6 +60,7 @@ func (e ErrResolveRevisit) String() string {
 }
 
 func (l *Loader) Load(ctx context.Context, warcId string) (gowarc.WarcRecord, error) {
+	log.Debug().Msg("loader load")
```
Comment (maeb) [inline internal/loader/loader.go:63]:
```suggestion
```
Stray debug log message..

File: internal/loader/loader.go [hunk 4]
```diff
@@ -79,13 +87,48 @@ func (l *Loader) Load(ctx context.Context, warcId string) (gowarc.WarcRecord, er
...
+			reqUrl := *l.ProxyUrl
+			reqUrl.Path = path.Join(reqUrl.Path, "id", warcRefersTo)
+
+			log.Debug().Msgf("attempt to get record from proxy url %s", reqUrl.String())
+			req, err := http.NewRequestWithContext(ctx, "GET", reqUrl.String(), nil)
+			if err != nil {
+				return nil, err
+			}
+
+			resp, err := http.DefaultClient.Do(req)
```
Comment (maeb) [inline internal/loader/loader.go:105]: Unecessary to create http.Client every time. Should be package global variable (see client internal/server/handlers/handlers.go for example).

File: internal/server/coreserver/contenthandler.go [hunk 5]
```diff
@@ -17,77 +17,50 @@
...
 }
 
 func (h contentHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	warcId, ok := mux.Vars(r)["id"]
-	if !ok {
-		http.NotFound(w, r)
+	params := httprouter.ParamsFromContext(r.Context())
+	warcId := params.ByName("id")
+	if warcId == "" {
+		http.Error(w, "400 bad request", http.StatusBadRequest)
```
Comment (maeb) [inline internal/server/coreserver/contenthandler.go:36]:
```suggestion
		http.Error(w, `missing required parameter "id"`, http.StatusBadRequest)
```

**Review by maeb — COMMENTED** [review]
Usual nitpicking :eyes:

File: internal/loader/loader.go (hunk 4 above)
Comment (Avokadoen) [reply internal/loader/loader.go:105]: Using default client does not create a new client as it is a http package wide client pointer:

client.go:
```go
// DefaultClient is the default Client and is used by Get, Head, and Post.
var DefaultClient = &Client{}
```

**Pull request merged**